test/csuite/CMakeLists.txt

```diff
@@ -287,15 +296,17 @@ define_c_test(
     TARGET test_wt6185_modify_ts
     SOURCES wt6185_modify_ts/main.c
     DIR_NAME wt6185_modify_ts
-    SMOKE
+    EXEC_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/wt6185_modify_ts/smoke.sh
+    ARGUMENTS $<TARGET_FILE:test_wt6185_modify_ts>
     DEPENDS "WT_POSIX"
 )
 
 define_c_test(
     TARGET test_wt6616_checkpoint_oldest_ts
     SOURCES wt6616_checkpoint_oldest_ts/main.c
     DIR_NAME wt6616_checkpoint_oldest_ts
-    SMOKE
+    EXEC_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/wt6616_checkpoint_oldest_ts/smoke.sh
+    ARGUMENTS $<TARGET_FILE:test_wt6616_checkpoint_oldest_ts>
```

Comment:
Looks like the value of `TARGET_FILE` (i.e test_wt6616_checkpoint_oldest_ts) equals that of `TARGET` for most of the tests (if not all). Is it syntactically possible to save the "duplication" with a variable reference?

Reply:
That's a good question, but that would unfortunately be against the intention of this PR. I intentionally moved this out of the `define_c_test` function. In this case the extra verbosity and duplication were intentional changes and are an important factor when making the specification of each test definition clear. In particular
- `ARGUMENTS` is an internal detail to the execution script or binary. As you noted `TARGET_FILE` is repeatedly used. This pattern is present since most smoke.sh scripts were updated to take the binary location as an argument (its a bit of temporary hack until autoconf is removed). _Thats not necessarily the case for all smoke.sh scripts_. Moving them into to `define_c_test` abstracts this important detail and assumes how arguments will be passed into the execution script. 
- `EXEC_SCRIPT` was exposed to introduce additional flexibility to define any custom wrapper script. Since it carries path information, where the script doesn't necessarily need to exist in the tests subdirectory, its unfortunately a bit difficult to simplify any further.

Reply:
Making sense to me.